### ISSUE TITLE
Six Lane Road Texture

### ISSUE THREAD

**Issue by TheColorRed** [Katalyst6/CSL.NetworkExtensions#16]
Hello, I would like to help you out with creating additional road types. I have started on a "Six Lane Avenue with Bus Lanes". I would like to modify a six lane avenue texture, and was wondering if you had one, or knew where the default one was that I could modify?

PS

Is there anything that I should know when creating a different type of road type?

**Comment by andreharv** [comment]
This might have already been done by the original devs.  Look at the PAX trailer for the 'After Dark' expansion at 0:17.  I think they have added bus lanes for many of their city roads.  If that was not what you had in mind, it would be nice to know more.

**Comment by TheColorRed** [comment]
Yeah that is pretty much it, but it doesn't necessarily have to be a bus lane, it could be a cargo lane, city services lane, etc.

**Comment by andreharv** [comment]
I see.  Another thing to consider is that Traffic++ has a feature that allows you to customize which vehicles use which lanes. Those bus lanes, I'm guessing, don't use any mechanics that are not already in the game now.  The bike lanes (green) do though.  Nonetheless, they will probably be customizable in the same way by Traffic++ upon their release.

**Comment by TheColorRed** [comment]
The city I live near has 2 lanes reserved (in order): 1 bus, 1 bike, 3+ everything else.

I think that it is something like this for those bus lanes:

```
lane.m_laneType = NetInfo.LaneType.PublicTransport;
```

for a bike lane, I think it would be something like this:

```
lane.m_vehicleType = VehicleInfo.VehicleType.Bicycle;
```

But I am not 100% sure, since there isn't any documentation.

**Comment by andreharv** [comment]
You are definitely on the right track!  and having a bus AND bicycle lane are totally possible.  You are right there is no clear documentation so if you have any questions, I don't know much, but I have been 'struggling' with this for a little while so just ask ;).
